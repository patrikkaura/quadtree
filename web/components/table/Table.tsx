import { QuadtreeData } from "@components/types";
import type { MRT_ColumnDef } from "material-react-table"; // If using TypeScript (optional, but recommended)
import MaterialReactTable from "material-react-table";
import React, { useMemo } from "react";

import { Sequence } from "./Sequence";

type PredictorResult = QuadtreeData["results"][0];

export type Props = {
  data: PredictorResult[];
};

export function Table({ data }: Props) {
  const columns = useMemo<MRT_ColumnDef<PredictorResult>[]>(
    () => [
      {
        accessorFn: (originalRow) => originalRow.interval[0],
        id: "start",
        header: "Start",
        size: 10,
      },
      {
        accessorFn: (originalRow) => originalRow.interval[1],
        id: "end",
        header: "End",
        size: 10,
      },
      {
        accessorFn: (originalRow) => originalRow.length,
        id: "length",
        header: "Length",
        size: 10,
      },
      {
        accessorFn: (originalRow) => originalRow.meanScore,
        id: "meanScore",
        header: "Avg. score",
        size: 10,
        Cell: ({ cell }) => cell.getValue<number>().toFixed(2),
      },
      {
        accessorFn: (originalRow) => originalRow.sequence,
        id: "sequence",
        header: "Sequence",
        size: 20,
        Cell: ({ cell, row }) => (
          <Sequence index={row.index} sequence={cell.getValue<string>()} />
        ),
      },
    ],
    []
  );

  return (
    <MaterialReactTable
      columns={columns}
      data={data}
      enableRowNumbers
      rowNumberMode="original"
      enableColumnFilters
      enableTopToolbar
      enableColumnActions={false}
      muiTableBodyRowProps={{ hover: true }}
      enableColumnOrdering
      enableRowSelection={false}
      enableDensityToggle={false}
      enableFullScreenToggle={false}
    />
  );
}
