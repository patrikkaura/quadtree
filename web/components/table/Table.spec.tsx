import "@testing-library/jest-dom";

import { render, screen } from "@testing-library/react";

import { Props, Table } from "./Table";

describe("Table", () => {
  let props: Props = {
    data: [
      {
        index: 0,
        interval: [0, 1],
        length: 999,
        meanScore: 0.2,
        scores: [0.1, 0.2, 0.3],
        sequence: "AATATAGGGGTATA",
      },
      {
        index: 1,
        interval: [10, 11],
        length: 10,
        meanScore: 0.5,
        scores: [0.1, 0.2, 0.3],
        sequence: "AATGGATAGGGGTATA",
      },
    ],
  };

  it("should render", async () => {
    const { container } = render(<Table {...props} />);

    expect(await screen.findByTestId("sequence-0")).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});
