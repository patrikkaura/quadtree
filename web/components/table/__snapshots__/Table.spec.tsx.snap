// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table should render 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation2 css-80pr5n-MuiPaper-root"
  >
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-dense css-1omuo8w-MuiToolbar-root"
    >
      <div
        class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
        style="min-height: 0px;"
      >
        <div
          class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
        >
          <div
            class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardInfo MuiAlert-standard css-leauvm-MuiPaper-root-MuiAlert-root"
              role="alert"
            >
              <div
                class="MuiAlert-message css-1pxa9xg-MuiAlert-message"
              >
                <div
                  class="MuiBox-root css-4sd9dj"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root css-1ag9q10"
      >
        <span />
        <div
          class="MuiBox-root css-1cjiko4"
        >
          <div
            class="MuiBox-root css-di3982"
          >
            <button
              aria-label="Show/Hide search"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
              data-mui-internal-clone-element="true"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="SearchIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-label="Show/Hide filters"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
              data-mui-internal-clone-element="true"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="FilterListIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 18h4v-2h-4v2zM3 6v2h18V6H3zm3 7h12v-2H6v2z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-label="Show/Hide columns"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
              data-mui-internal-clone-element="true"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="ViewColumnIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M14.67 5v14H9.33V5h5.34zm1 14H21V5h-5.33v14zm-7.34 0V5H3v14h5.33z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiTableContainer-root css-1vlry5a-MuiTableContainer-root"
    >
      <table
        class="MuiTable-root css-12xbor9-MuiTable-root"
      >
        <thead
          class="MuiTableHead-root css-1e3t2tg-MuiTableHead-root"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-head css-suriyt-MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-9l8l7c-MuiTableCell-root"
              colspan="1"
              scope="col"
            >
              <div
                class="Mui-TableHeadCell-Content MuiBox-root css-an1zfc"
              >
                <div
                  class="Mui-TableHeadCell-Content-Labels MuiBox-root css-4ng264"
                >
                  <div
                    class="Mui-TableHeadCell-Content-Wrapper MuiBox-root css-1habcj5"
                  >
                    #
                  </div>
                </div>
                <div
                  class="Mui-TableHeadCell-Content-Actions MuiBox-root css-epvm6"
                />
              </div>
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-ec4w4w-MuiTableCell-root"
              colspan="1"
              scope="col"
            >
              <div
                class="Mui-TableHeadCell-Content MuiBox-root css-an1zfc"
              >
                <div
                  class="Mui-TableHeadCell-Content-Labels MuiBox-root css-68rqdf"
                >
                  <div
                    class="Mui-TableHeadCell-Content-Wrapper MuiBox-root css-8uhtka"
                    title="Start"
                  >
                    Start
                  </div>
                  <span
                    aria-label="Unsorted"
                    class="MuiButtonBase-root MuiTableSortLabel-root css-f3gus1-MuiButtonBase-root-MuiTableSortLabel-root"
                    data-mui-internal-clone-element="true"
                    role="button"
                    tabindex="0"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-1vweko9-MuiSvgIcon-root-MuiTableSortLabel-icon"
                      data-testid="ArrowDownwardIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="m20 12-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                      />
                    </svg>
                  </span>
                </div>
                <div
                  class="Mui-TableHeadCell-Content-Actions MuiBox-root css-epvm6"
                >
                  <button
                    aria-label="Move"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-tqueyn-MuiButtonBase-root-MuiIconButton-root"
                    data-mui-internal-clone-element="true"
                    draggable="true"
                    tabindex="0"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="DragHandleIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M20 9H4v2h16V9zM4 15h16v-2H4v2z"
                      />
                    </svg>
                  </button>
                </div>
              </div>
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-ec4w4w-MuiTableCell-root"
              colspan="1"
              scope="col"
            >
              <div
                class="Mui-TableHeadCell-Content MuiBox-root css-an1zfc"
              >
                <div
                  class="Mui-TableHeadCell-Content-Labels MuiBox-root css-68rqdf"
                >
                  <div
                    class="Mui-TableHeadCell-Content-Wrapper MuiBox-root css-8uhtka"
                    title="End"
                  >
                    End
                  </div>
                  <span
                    aria-label="Unsorted"
                    class="MuiButtonBase-root MuiTableSortLabel-root css-f3gus1-MuiButtonBase-root-MuiTableSortLabel-root"
                    data-mui-internal-clone-element="true"
                    role="button"
                    tabindex="0"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-1vweko9-MuiSvgIcon-root-MuiTableSortLabel-icon"
                      data-testid="ArrowDownwardIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="m20 12-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                      />
                    </svg>
                  </span>
                </div>
                <div
                  class="Mui-TableHeadCell-Content-Actions MuiBox-root css-epvm6"
                >
                  <button
                    aria-label="Move"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-tqueyn-MuiButtonBase-root-MuiIconButton-root"
                    data-mui-internal-clone-element="true"
                    draggable="true"
                    tabindex="0"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="DragHandleIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M20 9H4v2h16V9zM4 15h16v-2H4v2z"
                      />
                    </svg>
                  </button>
                </div>
              </div>
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-ec4w4w-MuiTableCell-root"
              colspan="1"
              scope="col"
            >
              <div
                class="Mui-TableHeadCell-Content MuiBox-root css-an1zfc"
              >
                <div
                  class="Mui-TableHeadCell-Content-Labels MuiBox-root css-68rqdf"
                >
                  <div
                    class="Mui-TableHeadCell-Content-Wrapper MuiBox-root css-8uhtka"
                    title="Length"
                  >
                    Length
                  </div>
                  <span
                    aria-label="Unsorted"
                    class="MuiButtonBase-root MuiTableSortLabel-root css-f3gus1-MuiButtonBase-root-MuiTableSortLabel-root"
                    data-mui-internal-clone-element="true"
                    role="button"
                    tabindex="0"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-1vweko9-MuiSvgIcon-root-MuiTableSortLabel-icon"
                      data-testid="ArrowDownwardIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="m20 12-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                      />
                    </svg>
                  </span>
                </div>
                <div
                  class="Mui-TableHeadCell-Content-Actions MuiBox-root css-epvm6"
                >
                  <button
                    aria-label="Move"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-tqueyn-MuiButtonBase-root-MuiIconButton-root"
                    data-mui-internal-clone-element="true"
                    draggable="true"
                    tabindex="0"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="DragHandleIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M20 9H4v2h16V9zM4 15h16v-2H4v2z"
                      />
                    </svg>
                  </button>
                </div>
              </div>
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-ec4w4w-MuiTableCell-root"
              colspan="1"
              scope="col"
            >
              <div
                class="Mui-TableHeadCell-Content MuiBox-root css-an1zfc"
              >
                <div
                  class="Mui-TableHeadCell-Content-Labels MuiBox-root css-68rqdf"
                >
                  <div
                    class="Mui-TableHeadCell-Content-Wrapper MuiBox-root css-8uhtka"
                    title="Quadtree score"
                  >
                    Quadtree score
                  </div>
                  <span
                    aria-label="Unsorted"
                    class="MuiButtonBase-root MuiTableSortLabel-root css-f3gus1-MuiButtonBase-root-MuiTableSortLabel-root"
                    data-mui-internal-clone-element="true"
                    role="button"
                    tabindex="0"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-1vweko9-MuiSvgIcon-root-MuiTableSortLabel-icon"
                      data-testid="ArrowDownwardIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="m20 12-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                      />
                    </svg>
                  </span>
                </div>
                <div
                  class="Mui-TableHeadCell-Content-Actions MuiBox-root css-epvm6"
                >
                  <button
                    aria-label="Move"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-tqueyn-MuiButtonBase-root-MuiIconButton-root"
                    data-mui-internal-clone-element="true"
                    draggable="true"
                    tabindex="0"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="DragHandleIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M20 9H4v2h16V9zM4 15h16v-2H4v2z"
                      />
                    </svg>
                  </button>
                </div>
              </div>
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-ec4w4w-MuiTableCell-root"
              colspan="1"
              scope="col"
            >
              <div
                class="Mui-TableHeadCell-Content MuiBox-root css-an1zfc"
              >
                <div
                  class="Mui-TableHeadCell-Content-Labels MuiBox-root css-68rqdf"
                >
                  <div
                    class="Mui-TableHeadCell-Content-Wrapper MuiBox-root css-8uhtka"
                    title="Sequence"
                  >
                    Sequence
                  </div>
                  <span
                    aria-label="Unsorted"
                    class="MuiButtonBase-root MuiTableSortLabel-root css-f3gus1-MuiButtonBase-root-MuiTableSortLabel-root"
                    data-mui-internal-clone-element="true"
                    role="button"
                    tabindex="0"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-1vweko9-MuiSvgIcon-root-MuiTableSortLabel-icon"
                      data-testid="ArrowDownwardIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="m20 12-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                      />
                    </svg>
                  </span>
                </div>
                <div
                  class="Mui-TableHeadCell-Content-Actions MuiBox-root css-epvm6"
                >
                  <button
                    aria-label="Move"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-tqueyn-MuiButtonBase-root-MuiIconButton-root"
                    data-mui-internal-clone-element="true"
                    draggable="true"
                    tabindex="0"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="DragHandleIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M20 9H4v2h16V9zM4 15h16v-2H4v2z"
                      />
                    </svg>
                  </button>
                </div>
              </div>
            </th>
          </tr>
        </thead>
        <tbody
          class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-hover css-17rzbf3-MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-o0oj3w-MuiTableCell-root"
            >
              1
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1lasr7a-MuiTableCell-root"
            >
              0
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1lasr7a-MuiTableCell-root"
            >
              1
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1lasr7a-MuiTableCell-root"
            >
              999
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1lasr7a-MuiTableCell-root"
            >
              0.20
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1lasr7a-MuiTableCell-root"
            >
              <span
                data-testid="sequence-0"
              >
                A
                A
                T
                A
                T
                A
                <span
                  style="font-size: 20px; font-weight: 500; border-bottom: 1px solid black; color: rgb(255, 94, 124);"
                >
                  G
                  G
                  G
                  G
                </span>
                T
                A
                T
                A
              </span>
            </td>
          </tr>
          <tr
            class="MuiTableRow-root MuiTableRow-hover css-17rzbf3-MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-o0oj3w-MuiTableCell-root"
            >
              2
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1lasr7a-MuiTableCell-root"
            >
              10
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1lasr7a-MuiTableCell-root"
            >
              11
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1lasr7a-MuiTableCell-root"
            >
              10
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1lasr7a-MuiTableCell-root"
            >
              0.50
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1lasr7a-MuiTableCell-root"
            >
              <span
                data-testid="sequence-1"
              >
                A
                A
                T
                <span
                  style="font-size: 20px; font-weight: 500; border-bottom: 1px solid black; color: rgb(255, 94, 124);"
                >
                  G
                  G
                </span>
                A
                T
                A
                <span
                  style="font-size: 20px; font-weight: 500; border-bottom: 1px solid black; color: rgb(255, 94, 124);"
                >
                  G
                  G
                  G
                  G
                </span>
                T
                A
                T
                A
              </span>
            </td>
          </tr>
        </tbody>
        <tfoot
          class="MuiTableFooter-root css-g5ysz2-MuiTableFooter-root"
        />
      </table>
    </div>
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-dense css-g2tram-MuiToolbar-root"
    >
      <div
        class="MuiBox-root css-1llu0od"
      >
        <span />
        <div
          class="MuiBox-root css-qpxlqp"
        >
          <div
            class="MuiTablePagination-root css-16253fi-MuiTablePagination-root"
          >
            <div
              class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular MuiTablePagination-toolbar css-78c6dr-MuiToolbar-root-MuiTablePagination-toolbar"
            >
              <div
                class="MuiTablePagination-spacer css-1psng7p-MuiTablePagination-spacer"
              />
              <p
                class="MuiTablePagination-selectLabel css-pdct74-MuiTablePagination-selectLabel"
                id=":rf:"
              >
                Rows per page
              </p>
              <div
                class="MuiInputBase-root MuiInputBase-colorPrimary css-6vypie-MuiInputBase-root-MuiTablePagination-select"
              >
                <div
                  aria-expanded="false"
                  aria-haspopup="listbox"
                  aria-labelledby=":rf: :re:"
                  class="MuiSelect-select MuiTablePagination-select MuiSelect-standard MuiInputBase-input css-194a1fa-MuiSelect-select-MuiInputBase-input"
                  id=":re:"
                  role="button"
                  tabindex="0"
                >
                  10
                </div>
                <input
                  aria-hidden="true"
                  class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                  tabindex="-1"
                  value="10"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiTablePagination-selectIcon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
                  data-testid="ArrowDropDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </div>
              <p
                class="MuiTablePagination-displayedRows css-levciy-MuiTablePagination-displayedRows"
              >
                1-2 of 2
              </p>
              <div
                class="MuiTablePagination-actions"
              >
                <button
                  aria-label="Go to previous page"
                  class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
                  disabled=""
                  tabindex="-1"
                  title="Go to previous page"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="KeyboardArrowLeftIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                    />
                  </svg>
                </button>
                <button
                  aria-label="Go to next page"
                  class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
                  disabled=""
                  tabindex="-1"
                  title="Go to next page"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="KeyboardArrowRightIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
